### ISSUE TITLE
Thunderbird version > 40 does not show "launch keepass" button

### ISSUE THREAD

**Issue by benjiof76** [kee-org/KeeFox#656]
http://i.imgur.com/JnSLkLR.png

**Comment by dlech** [comment]
Which versions of Thunderbird and KeeFox are you using?

Are there any errors in the Error Console?

**Comment by benjiof76** [comment]
thunderbird 45.0
Keefox 1.6.0

Yes, :
http://i.imgur.com/DFKhmyh.png
http://i.imgur.com/H9fyJjt.png

**Comment by benjiof76** [comment]
without firewall :
http://i.imgur.com/QVuGbzt.png
http://i.imgur.com/P2rj3fd.png

**Comment by dlech** [comment]
There are some fixes for Thunderbird 40+ that are currently only in Keefox 1.6.1b1. Please try that version. 

https://addons.mozilla.org/en-US/firefox/addon/keefox/versions/beta

**Comment by benjiof76** [comment]
always the same, I will send you the error console

**Comment by benjiof76** [comment]
he asks me keepass when sending the message but not the opening of 
thunderbird

**Comment by benjiof76** [comment]
he asks me keepass when sending the message but not the opening of thunderbird
http://i.imgur.com/JutCsQa.png
http://i.imgur.com/aGsCezI.png
http://i.imgur.com/aPBI4Bg.png
http://i.imgur.com/az6bOdw.png

**Comment by dlech** [comment]
Did you upgrade to v1.6.1b1? It won't work with IMAP accounts otherwise.

**Comment by benjiof76** [comment]
Yes.

**Comment by benjiof76** [comment]
I abandoned the idea of solving this problem. I returned to the previous release (Thunderbird 38.7.2) with keefox 1.6. If you want me to continue with the latest version of Thunderbird, let me know.

**Comment by benjiof76** [comment]
I abandoned the idea of solving this problem. I returned to the previous release (Thunderbird 38.7.2) with keefox 1.6. If you want me to continue solving this problem with the latest version of Thunderbird, to help the community, let me know.

**Comment by dlech** [comment]
You have confirmed that SMTP does work. Which type of accounts are not working? POP? IMAP?

**Comment by MarcoWilli** [comment]
Same for me here since upgrade to v45 today. Gonna look for how to downgrade... :-/

**Comment by benjiof76** [comment]
Choose your language and download it : http://ftp.mozilla.org/pub/thunderbird/releases/38.7.2/win32/

**Comment by dlech** [comment]
v1.6.1b1 seems to be working for most people. For those that it is not, what language/translation are you using? @benjiof76, it looks like you are using French, yes?

**Comment by benjiof76** [comment]
Yes ! ^^

**Comment by Sardtok** [comment]
The beta doesn't work for me. My Thunderbird is Norwegian, although part of the dialog for the login shows in English.
I see a Timer.jsm error whenever the login dialog pops up. I.e. if I cancel the login and try to open the inbox again, the Timer.jsm error will re-occur. It says that aCallback.apply is not a function. I see the same errors in benjiof76's screenshots above. Not sure if you use timeouts, but if you do, I guess that's worth having a look at.

I'm using IMAP, if you were wondering.

**Comment by ggeorgg** [comment]
Great! 
After the update of thunderbird to v45.0 my auto fill out didn't work anymore and the start keepass button had not been displayed.
I simply installed the beta version and it works again! Thanks for your support :+1:

**Comment by Sardtok** [comment]
The issue magically disappeared for me after the 2.33 release of KeePass the other day.

**Comment by HannsDieter** [comment]
I have the same problem. 
Just updated to Thunderbird 45. and KeePass 2.33.
There is no start keepass button and auto fill isn't working. I have the german version of Thunderbird. 
(It is working fine with firefox though)

**Comment by kevinthekingsson** [comment]
I have FireFox 46.0.1 and Thunderbird 45.0.1. My system, which is standard English, has been working fine with Keefox for over a year. But after an automatic update last week, Thunderbird stopped prompting me to open KeePass and, even when it was open, would not use it; so for about a week now, I've had to enter all my email passwords by hand. Firefox, however, continues to recognise and use KeeFox. I've tried removing and reinstalling KeeFox, flushing the KeePass plugins cache, etc., but no joy. Tried deinstalling Keefox and installing v1.6.1b1 as suggested above: but still not recognised by Thunderbird. As there doesn't seem to be any way of explicitly telling Thunderbird to install v1.6.1b1 I've tried both installing to Firefox over the existing Thunderbird setup and removing the existing Keefox plugin before reinstalling to Firefox. But nothing seems to work. Any further suggestions?

**Comment by Sardtok** [comment]
After a reboot and restarting Thunderbird (second launch since the KeePass update), the KeePass integration stopped working again. :(

**Comment by HannsDieter** [comment]
Now also a problem on my Ubuntu machine after Thunderbird update to 45.2. 
(My system is in english)

Is this issue been worked on? (Taken that Keefox is not at all working since April)

**Comment by kevinthekingsson** [comment]
This is still a problem for me also. I'm getting very tired of having to find all 6 of my email accounts and then use autotype to enter the passwords every time I start up Thunderbird. So I am starting to think about alternatives...

**Comment by dlech** [comment]
Has anyone tried [logging](https://github.com/luckyrat/KeeFox/wiki/en-%7C-Options-%7C-Logging) to see if there are any useful error messages?

**Comment by raymondBourges** [comment]
Same issue with Thunderbird 45.2 on my Ubuntu. My log file:

Wed Aug 03 2016 11:37:28 GMT+0200 (CEST):DEBUG: Observed an event: [xpconnect wrapped nsIPrefBranch],nsPref:changed,logLevel
Wed Aug 03 2016 11:37:33 GMT+0200 (CEST):INFO: Window shutting down...
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):Logging system initialised at Wed Aug 03 2016 11:37:36 GMT+0200 (CEST)
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: KeeFox initialising
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: KeePass install location found in preferences: /usr/lib/keepass2
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: KeePass EXE found in correct location.
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: Looking for the KeePassRPC plugin plgx
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: KeePassRPC plgx found in specified or default location.
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: Mono install location found in preferences: /usr/bin/mono
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: Mono executable found in correct location.
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: Checking and updating KeePassRPC installation settings
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: KeePass is not running or the connection might be established in a second...
Wed Aug 03 2016 11:37:36 GMT+0200 (CEST):INFO: KeeFox initialised OK although the connection to KeePass may not be established just yet...
Wed Aug 03 2016 11:37:37 GMT+0200 (CEST):INFO: Testing to see if we've already established whether KeePassRPC is connected.
Wed Aug 03 2016 11:37:38 GMT+0200 (CEST):INFO: Websocket connection opened
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: KeePass install location found in preferences: /usr/lib/keepass2
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: KeePass EXE found in correct location.
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: Looking for the KeePassRPC plugin plgx
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: KeePassRPC plgx found in specified or default location.
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: Mono install location found in preferences: /usr/bin/mono
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: Mono executable found in correct location.
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: Checking and updating KeePassRPC installation settings
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: KeePass is not running or the connection might be established in a second...
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: At least one database is open.
Wed Aug 03 2016 11:37:39 GMT+0200 (CEST):INFO: KeeFox feels very refreshed now.

**Comment by dlech** [comment]
Thanks for the log. Which version of KeeFox and KeePassRPC are you using?

**Comment by raymondBourges** [comment]
KeeFox 1.6.0
KeePassRPC 1.6.0.0

**Comment by dlech** [comment]
KeeFox 1.6.0 is known to not work with Thunderbird >=40. Please use the latest beta (currently 1.6.3b1).

Also, please check for errors in the error console. (Tools > Error Console).

It seems as if the commonDialog overlay is not being loaded at all for some reason, but there is nothing in the logs indicating a problem.

**Comment by raymondBourges** [comment]
Ok with 1.6.3b1. Thanks

**Comment by kevinthekingsson** [comment]
Success at last! But not until I finally sussed out how to force
      FireFox and Thunderbird to perform the upgrade, which took me
      several hours to discover. For others who normally only use
      official upgrades rather than betas, here's how...
    Firefox.
    By default, Firefox will not allow you to install anything except
      officially released upgrades. To install the beta version, do as
      follows:
    
      In the Firefox address bar, type about:config and
        press Enter.
      Click the 'I'll be careful...' button
      Near the bottom of the next screen find the entry
        'xp.signatures.required' entry. Right click on it and select
        'Toggle' to change the value to' false'.
      Now go to
        https://github.com/luckyrat/KeeFox/releases/tag/v1.6.3b1 (or
        whatever the latest version is)
      Right click on the <version no>.xpi download link and 
      Click on the <version no>.xpi download link and confirm
        that you really do want to install it.
      Follow the prompts to complete the upgrade process (including
        entering a new security key) and that should sort out Firefox
        upgraded to the latest beta.
      I was going to reset 'xp.signatures.required' back to its
        default value: but this displayed a message suggesting it was
        about to disable Keefox again: so I decided not to try that for
        now...)
    
    Thunderbird
    At first sight, there seems to be no way of forcing Thunderbird
      to accept a non-standard upgrade. To do this, proceed as follows:
    
      In Firefox, go back to
        https://github.com/luckyrat/KeeFox/releases/tag/v1.6.3b1 (or
        whatever the latest version is)
      Right click on the <version no>.xpi download link and
        select 'Copy link location.'
      
      Now select Tools/Add-ons/Extensions from the Thunderbird main
        menu.
      Click the 'Tools' cog symbol next to the search bar, and
        select 'Install Add-on from File.' A 'Select Add-on' dialog will
        appear.
      Paste the link copied at step 2 into the 'File Name' box and
        click 'Open.'
      The system may chew this over for a while, and you may have to
        press 'Open' a second time: but eventually you should be asked
        to confirm that you are really serious about this.
      Follow the remaining prompts (including entering a new
        security key) and you should end up with a working beta version.
    
    It worked for me, anyway....=-O
    
    On 04-Aug-16 9:00 AM, Raymond Bourges
      wrote:
    
    
      Ok with 1.6.3b1. Thanks
      —
        You are receiving this because you commented.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/luckyrat/KeeFox","title":"luckyrat/KeeFox","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/modules/aws/aws-bg.jpg","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/luckyrat/KeeFox"}},"updates":{"snippets":[{"icon":"PERSON","message":"@raymondBourges in #656: Ok with 1.6.3b1. Thanks"}],"action":{"name":"View Issue","url":"https://github.com/luckyrat/KeeFox/issues/656#issuecomment-237481202"}}}

**Comment by ggeorgg** [comment]
It's not working for me. 
Keefox v1.6.3b1 not working with Thunderbird Version 47 and unfortunatley not working anymore with Thunderbird Version 45.2 like it was before (See [my post from april](https://github.com/luckyrat/KeeFox/issues/656#issuecomment-215947780))

Following Error messages occurs when Keefox is v1.6.3b1 is enabled in Thunderbird V45.2
![errorlogthv45withkfv163b1](https://cloud.githubusercontent.com/assets/13244552/17492230/2a1eb2f0-5dab-11e6-9f8e-2484c6021ae0.jpg)

Without Keefox enabled the error log is displayed as followed:
![errorlogthv45withoutkfv163b1](https://cloud.githubusercontent.com/assets/13244552/17492254/426697ec-5dab-11e6-8125-351b591e62f0.jpg)

The same error messages occur with Thunderbird V47.0beta.

If i will ever get this to work, I think I will disable automatic updates...

**Comment by boldt** [comment]
FYI: For me, Keefox 1.6.3b1 works with Thunderbrid 45.2.0

**Comment by dlech** [comment]
I am still unable to reproduce this problem. I have just tried French language on Thunderbird 45.4.0 on Linux with KeeFox 1.6.4.

If anyone else is still having trouble, please give as much information as possible about your installation.

**Comment by dmuneram** [comment]
Hello, 
I'm having this issue. Launch KeePass button is missing.
- Ubuntu 16.10 (English)
- Thunderbird 45.3.0
- KeeFox 1.6.4

Timestamp: 13/01/17 08:58:25
Error: TypeError: aCallback.apply is not a function
Source File: resource://gre/modules/Timer.jsm
Line: 30
Timestamp: 13/01/17 08:58:26
Warning: Content Security Policy: Directive 'frame-src' has been deprecated. Please use directive 'child-src' instead.
Timestamp: 13/01/17 08:58:26
Error: The connection was refused when attempting to contact ws://127.0.0.1:12546/.
Source File: resource://kfmod/session.js
Line: 154
Timestamp: 13/01/17 08:58:34
Error: uncaught exception: 2147746065
Source File: resource:///modules/activity/autosync.js
Line: 210

If you need anymore info, let me know.

**Comment by dlech** [comment]
Due to the recent split between Thunderbird and Firefox extensions, we have started a fork of KeeFox specifically for Thunderbird. We are calling this new extension KeeBird and the code and issue tracking are now in a new GitHub repository. If anyone is still interested in getting this issue resolved, please let us know by opening a new issue at https://github.com/kee-org/keebird/issues and copy/paste any relevant information (or at least link to this issue).
